### PR TITLE
Issues resource with Comments and Attachments subresources were added

### PR DESCRIPTION
**Pull request by promoisha**
Issues resource with Comments and Attachments subresources were added under theAPI resource.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by olydis** [comment]
@ravbhatnagar new operations

**Comment by promoisha** [comment]
@solankisamir

**Comment by promoisha** [comment]
Hi, @ravbhatnagar! Could you please take a look? Thanks!

**Comment by ravbhatnagar** [comment]
Looks good. Signing off from ARM side.

**Comment by promoisha** [comment]
@olydis Don't merge this PR, as the API is not out yet.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json
Comment (olydis) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json]: Just a non-technical suggestion: I'd not put "Put" in the name of this class, it's a technical detail of the API and not helpful to SDK users. Unfortunately, I thing there is no big consensus for naming these things, but I've seen things like "...Setting" or "...Parameters" before (where "..." is the underlying resource type, i.e. `IssueAttachmentContract` in this case). Maybe you can take a quick look at other "put" operations in this RP to find the common naming strategy 🙂

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json
Comment (olydis) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json]: Applies to all `...Collection` types: Could you mark all properties as `readOnly: true`?

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json
Comment (olydis) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json]: not used anywhere - thanks to the example you provided for the put operation, this was caught by our tooling 😃 Pretty sure you wanted to use this schema for `IssueAttachmentPutContract`'s `properties`. (My naming comment regarding using "Put" in the name of this type of course also applies for this type)

**Review by olydis — CHANGES_REQUESTED** [review]
Looking good overall, just one likely bug and some minor suggestions! 🙂 On which date is this API going live?

**Comment by olydis** [comment]
@promoisha please respond to outstanding comments/questions. Will otherwise close this PR due to inactivity (can be reopened once you are ready to move forward).

**Comment by olydis** [comment]
@salameer @promoisha Closing due to inactivity. Feel free to reopen once you're ready to address outstanding issues.

**Comment by olydis** [comment]
@promoisha Is this API already public/live or when is that going to happen? Just so I can merge at the right point in time 🙂

**Review by olydis — APPROVED** [review]
LGTM! Will merge once you confirmed whether/when this is live 🙂

**Comment by promoisha** [comment]
@olydis it's hitting the production right now, so I assume it's safe to merge :). Thanks a lot!

**Comment by olydis** [comment]
@promoisha Thank you!

**Pull request merged**
